Src/main.c

```diff
@@ -102,6 +102,11 @@ int main(void)
 
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
+  
+  HAL_ADC_Start(&hadc); //Laye - starts ADC
+  uint32_t converted_val = 0; //variable for ADC converted val - Laye
+  uint32_t compare_val = 0; //variable for compare register - Laye
```

Comment:
32 bits is probably more than what's necessary for these variables. Using uint16_t would save some memory.